rfcs/0003-validations-tdd/README.md

```diff
@@ -0,0 +1,151 @@
+
+## Background
+
+### Current Process
+
+**Step 1: Test Case Definition**  
+Subject Matter Experts (SMEs) provide desired logic and informal test scenarios in written form. These typically include household characteristics such as:
+
+- Household composition (number of people, relationships, ages)
+- Income sources and amounts
+- Existing benefit enrollments
+- Geographic location and other eligibility factors
+
+**Step 2: Initial Development**  
+Engineers implement the new program's eligibility logic and benefit calculation rules in their local development environment based on the SME requirements.
+
+**Step 3: Manual Testing via UI**  
+Engineers manually recreate each SME-provided test scenario by clicking through the screener interface, entering all household details, income information, and benefit history to verify their implementation produces the expected results.
+
+**Step 4: Iterative Development**  
+As engineers refine the program logic, they must repeatedly navigate through the entire screener UI for each test case to validate their changes. This creates a significant time burden, especially when testing multiple scenarios or making frequent adjustments.
+
+**Step 5: Staging Validation**  
+Once code is merged to staging, SMEs manually execute the same test scenarios through the screener interface to create official validations. These validations serve as regression detection benchmarks for future changes.
+
+### Problem Statement
+
+Our current workflow is heavily dependent on manual UI interaction for both developers and SMEs. This creates several inefficiencies:
+
+For Developers:
+
+- Significant time overhead clicking through screener forms for each test iteration
+- Difficulty testing edge cases and boundary conditions due to manual setup burden
+- Reduced likelihood of comprehensive testing during active development
+- Slower feedback loops when refining eligibility logic
+
+For SMEs:
+
+- Duplicate effort recreating the same scenarios they originally documented
+- Limited ability to provide structured, reusable test cases
+
+### Desired Solution
+
+We need a streamlined approach that:
+
+- Enables faster developer testing by reducing manual screener navigation during development iterations
+- Empowers SMEs to create structured, reusable test cases that can be programmatically executed across development, staging, and production environments
+- Bridges the gap between written requirements and executable tests to reduce translation errors and duplication of effort
+
+## Proposal
+
+### Overview
+
+SMEs create test cases by specifying screener inputs and expected outputs in JSON, which are attached to a ticket before implementation begins. An engineer runs a script to import these test cases as validations that can be used to verify program logic during development, as well as in staging and production environments.
```

Comment:
I wonder about JSON as the format for SMEs to specify their test cases. It feels like it could be closer to the technical side than it needs to be. The cost of translating human-readable test cases into valid JSON for the proposed script via AI seems small, and I think it potentially lower the lift for our SMEs.

What do you all think?